app/controllers/course/survey/responses_controller.rb

```diff
@@ -28,11 +28,22 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
-    if @response.update_attributes(response_update_params)
-      render_response_json
-    else
-      render json: { errors: @response.errors }, status: :bad_request
+    if params[:response][:unsubmit]
```

Comment:
😄  @jchiam sorry didn't notice your comment 

>Not sure if I'm doing the params filtering right.
I have an example of param filtering above